### ISSUE TITLE
Compilation error on develop branch

### ISSUE THREAD

**Issue by AugustinLu** [OrderN/CONQUEST-release#287]
From last week, compiling CONQUEST from the develop branch leads to an error as shown below:

```
mpif90 -O3 -I/home/augustin/local/lib/libxc-6.2.2/include -I/usr/local/include  -fallow-argument-mismatch -c S_matrix_module.f90
S_matrix_module.f90:275:41:

  275 |     call PAO_or_gradPAO_to_grid(atomfns, evaluate_pao, 0)
      |                                         1
Error: Interface mismatch in dummy procedure ‘evaluate’ at (1): Type mismatch in argument 'm' (INTEGER(4)/REAL(8))
make: *** [Makefile:85: S_matrix_module.o] Error 1
```

This seems related some calls in the **PAO_or_gradPAO_to_grid** subroutine.

Does anyone know where this comes from?

Thank you very much.

**Comment by davidbowler** [comment]
Yes, there is an error in the code; it will be fixed in the next day or two.  Please revert to a version of develop from a week or so ago.

Dave

On 20/11/2023, 07:52, "Augustin Lu" ***@***.***> wrote:


⚠ Caution: External sender


From last week, compiling CONQUEST from the develop branch leads to an error as shown below:

mpif90 -O3 -I/home/augustin/local/lib/libxc-6.2.2/include -I/usr/local/include  -fallow-argument-mismatch -c S_matrix_module.f90

S_matrix_module.f90:275:41:



  275 |     call PAO_or_gradPAO_to_grid(atomfns, evaluate_pao, 0)

      |                                         1

Error: Interface mismatch in dummy procedure ‘evaluate’ at (1): Type mismatch in argument 'm' (INTEGER(4)/REAL(8))

make: *** [Makefile:85: S_matrix_module.o] Error 1

This seems related some calls in the PAO_or_gradPAO_to_grid subroutine.

Does anyone know where this comes from?

Thank you very much.

**Comment by AugustinLu** [comment]
Thank you for your prompt reply. I will use an older version in the meantime.

Augustin
